ReactAndroid/src/main/java/com/facebook/react/ReactRootView.java

```diff
@@ -471,6 +472,87 @@ public void setAppProperties(@Nullable Bundle appProperties) {
     }
   }
 
+  private static class RootViewInsets {
+    float top;
+    float right;
+    float bottom;
+    float left;
+
+    RootViewInsets(float top, float right, float bottom, float left) {
+      this.top = top;
+      this.right = right;
+      this.bottom = bottom;
+      this.left = left;
+    }
+  }
+
+  private RootViewInsets getSafeAreaInsets() {
+    // Window insets are parts of the window that are covered by system views (status bar,
+    // navigation bar, notches). There are no apis the get these values for android < M so we
+    // do a best effort polyfill.
+    // TODO: Use the DisplayCutout api when we target sdk 28+
+    RootViewInsets windowInsets;
+    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
+      WindowInsets insets = getRootWindowInsets();
+      windowInsets = new RootViewInsets(
+        insets.getSystemWindowInsetTop(),
+        insets.getSystemWindowInsetRight(),
+        insets.getSystemWindowInsetBottom(),
+        insets.getSystemWindowInsetLeft());
+    } else {
+      int rotation =
+        ((WindowManager) getContext().getSystemService(Context.WINDOW_SERVICE))
+          .getDefaultDisplay().getRotation();
+      int statusBarHeight = 0;
+      int resourceId = getResources().getIdentifier("status_bar_height", "dimen", "android");
+      if (resourceId > 0) {
+        statusBarHeight = getResources().getDimensionPixelSize(resourceId);
+      }
+      int navbarHeight = 0;
+      resourceId = getResources().getIdentifier("navigation_bar_height", "dimen", "android");
+      if (resourceId > 0) {
+        navbarHeight = getResources().getDimensionPixelSize(resourceId);
+      }
+
+      windowInsets = new RootViewInsets(
+        statusBarHeight,
+        rotation == Surface.ROTATION_90 ? navbarHeight : 0,
+        rotation == Surface.ROTATION_0 || rotation == Surface.ROTATION_180 ? navbarHeight : 0,
+        rotation == Surface.ROTATION_270 ? navbarHeight : 0);
+    }
+
+    // Calculate the part of the root view that overlaps with window insets.
+    int[] windowLocation = new int[2];
+    getLocationInWindow(windowLocation);
+    DisplayMetrics screenMetrics = DisplayMetricsHolder.getScreenDisplayMetrics();
+    windowInsets.top = Math.max(windowInsets.top - windowLocation[1], 0);
```

Comment:
This seem inconsistent with iOS. When view occupies bottom half of the screen on iOS the top inset will be 0 whereas on Android it will be `windowHeight/2`

(same applies to left, bottom and right insets)

Reply:
Are you sure about this?

If the root view covers the bottom half of the screen then it would be:
Lets say that the status bar is visible and `windowInsets.top` = 25 initially

`Math.max(25 - ( windowHeight / 2), 0) -> Math.max(-somebignumber, 0) -> 0`

Let me know if there is something that could make the code clearer.

Reply:
ah, sorry I think you are right. Please ignore this comment